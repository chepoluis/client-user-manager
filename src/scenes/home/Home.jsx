import { Box } from "@mui/material";
import Header from "../../components/Header";

const Home = () => {
  return (
    <Box m="20px">
      <Box display="flex" justifyContent="space-between" alignItems="center">
        <Header title="Welcome" subtitle=":D" />
      </Box>
    </Box>
  );
};

export default Home;
