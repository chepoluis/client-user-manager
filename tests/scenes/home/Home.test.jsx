import { render, screen } from "@testing-library/react";
import Home from "../../../src/scenes/home/Home";
import { ThemeProvider, createTheme } from "@mui/material/styles";
import "@testing-library/jest-dom/extend-expect";

const theme = createTheme();

describe("tests in <Home />", () => {
  test("should match the snapshot", () => {
    const { asFragment } = render(
      <ThemeProvider theme={theme}>
        <Home />
      </ThemeProvider>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test("should render the component correctly", () => {
    render(<Home />);

    expect(screen.getByText(/Welcome/i)).toBeInTheDocument();
    expect(
      screen.getByText(/Start managing Users\/Accounts\/Teams with ease./i)
    ).toBeInTheDocument();

    const image = screen.getByAltText(/Cute puppy at the office/i);
    expect(image).toBeInTheDocument();
  });
});
