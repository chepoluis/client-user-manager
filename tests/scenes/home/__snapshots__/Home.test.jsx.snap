// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests in <Home /> should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-zwiamn"
  >
    <div
      class="MuiBox-root css-69i1ev"
    >
      <div
        class="MuiBox-root css-b5mc0i"
      >
        <div
          class="MuiBox-root css-0"
        >
          <h2
            class="MuiTypography-root MuiTypography-h2 css-be5vtk-MuiTypography-root"
          >
            Welcome
          </h2>
          <h5
            class="MuiTypography-root MuiTypography-h5 css-b1sixc-MuiTypography-root"
          >
            Start managing Users/Accounts/Teams with ease.
          </h5>
        </div>
        <div
          class="MuiBox-root css-0"
        />
      </div>
    </div>
    <img
      alt="Cute puppy at the office"
      class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-o69gx8-MuiCardMedia-root"
      src="../../../assets/cute_dog.png"
      style="display: flex; position: absolute; top: 50%; left: 50%; margin-top: 30px; transform: translate(-50%, -50%); width: 200px; height: 200px; border-radius: 20px;"
    />
  </div>
</DocumentFragment>
`;
